src/back-end/lib/resources/avatar.ts

```diff
@@ -0,0 +1,72 @@
+
+import * as crud from 'back-end/lib/crud';
+import { Connection } from 'back-end/lib/db';
+import * as permissions from 'back-end/lib/permissions';
+import FileResource, { CreateRequestBody, ValidatedCreateRequestBody } from 'back-end/lib/resources/file';
+import { SupportedRequestBodies, SupportedResponseBodies } from 'back-end/lib/types';
+import { CreateValidationErrors } from 'shared/lib/resources/file';
+import { Session } from 'shared/lib/resources/session';
+import { getInvalidValue, invalid, isValid, valid, Validation } from 'shared/lib/validation';
+import * as fileValidation from 'shared/lib/validation/file';
+
+type Resource = crud.Resource<
+  SupportedRequestBodies,
+  SupportedResponseBodies,
+  CreateRequestBody,
+  ValidatedCreateRequestBody,
+  CreateValidationErrors,
+  null,
+  null,
+  null,
+  null,
+  null,
+  null,
+  null,
+  Session,
+  Connection
+>;
+
+const resource: Resource = {
+  routeNamespace: 'avatars',
+
+  readOne: FileResource.readOne,
```

Comment:
Let's remove this and simply use `/api/files/:id` to read one file. This resource can just handle creating an avatar.